### PR TITLE
[SPARK-10482] [ML] Add Python interface for ml.CountVectorizer

### PR DESCRIPTION
**Pull request by hhbyyh**
jira: https://issues.apache.org/jira/browse/SPARK-10482

Add Python interface for feature transformer: ml.CountVectorizer

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -167,6 +168,134 @@ def getSplits(self):
 
 
 @inherit_doc
+class CountVectorizer(JavaEstimator, HasInputCol, HasOutputCol):
+    """
+    Extracts a vocabulary from document collections and generates a [[CountVectorizerModel]],
+    which converts text documents to sparse vectors of token counts.
+
+    >>> df = sentenceData = sqlContext.createDataFrame([(0, ["a", "b", "c"]),
```
Comment (mengxr) [inline python/pyspark/ml/feature.py]: remove `sentenceData = `

File: python/pyspark/ml/feature.py [hunk 2]
```diff
@@ -167,6 +168,134 @@ def getSplits(self):
...
+    ... (1, ["a", "b", "b", "c", "a"])], ["label", "raw"])
```
Comment (mengxr) [inline python/pyspark/ml/feature.py]: The following style might be better:

~~~python
df = sqlContext.createDataFrame(
    [(0, ["a", "b", "c"]), (1, ["a", "b", "b", "c", "a"])],
    ["label", "raw"])
~~~

File: python/pyspark/ml/feature.py [hunk 3]
```diff
@@ -167,6 +168,134 @@ def getSplits(self):
...
+                  "frequency/count less than the given threshold are ignored. If this is an " +
+                  "integer >= 1, then this specifies a count (of times the term must appear in" +
+                  " the document); if this is a double in [0,1), then this specifies a " +
+                  "fraction (out of the document's token count). Note that the parameter is " +
+                  "only used in transform of CountVectorizerModel and does not affect fitting.")
+
+    @keyword_only
+    def __init__(self, minDF=1.0, minTF=1.0, vocabSize=1 << 18, inputCol=None, outputCol=None):
+        """
+        __init__(self, minDF=1.0, minTF=1.0, vocabSize=1<<18, inputCol=None, outputCol=None)
```
Comment (mengxr) [inline python/pyspark/ml/feature.py]: `1 << 18`

File: python/pyspark/ml/feature.py [hunk 4]
```diff
@@ -167,6 +168,134 @@ def getSplits(self):
...
+
+class CountVectorizerModel(JavaModel):
+    """
+    Model fitted by CountVectorizer. Converts a text document to a sparse vector of token counts.
+    """
+
+    @property
+    def vocabulary(self):
+        """
+        An Array over terms. Only the terms in the vocabulary will be counted.
```
Comment (mengxr) [inline python/pyspark/ml/feature.py]: `array`

**Comment by mengxr** [comment]
LGTM except some minor issues

**Comment by holdenk** [comment]
This seems to do the same work as the outstanding PR https://github.com/apache/spark/pull/8561

**Comment by mengxr** [comment]
@holdenk Yes, I just noticed it. Could you merge some changes in this PR into yours? I think the doctest from @hhbyyh is better and the default values are specified correctly in this PR. I will make a pass after.

@hhbyyh Since this duplicates #8561, do you mind closing this PR? You can check opening PRs at https://spark-prs.appspot.com/#mllib.

**Comment by holdenk** [comment]
Ok, I'll merge in the doc tests.

**Comment by hhbyyh** [comment]
@mengxr Sorry for the extra effort during review.

**Pull request closed without merging**